src/app/AttributeCache.cpp

```diff
@@ -219,5 +219,75 @@ CHIP_ERROR AttributeCache::GetStatus(const ConcreteAttributePath & path, StatusI
     return CHIP_NO_ERROR;
 }
 
+struct compare
+{
+    bool operator()(const ConcreteDataAttributePathWithSize & x, const ConcreteDataAttributePathWithSize & y) const
+    {
+        return x.mSize < y.mSize;
+    }
+};
+
+uint32_t AttributeCache::OnUpdateDataVersionFilterList(DataVersionFilterIBs::Builder & aDataVersionFilterIBsBuilder,
+                                                       const Span<DataVersionFilter> & aDataVersionFilters)
+{
+    uint32_t number = 0;
+    CHIP_ERROR err  = CHIP_NO_ERROR;
+    TLV::TLVWriter backup;
+
+    std::set<ConcreteDataAttributePathWithSize, compare> clusterSet;
+    for (auto path = mChangedAttributeSet.begin(); path != mChangedAttributeSet.end(); path++)
```

Comment:
Why is this using the `mChangedAttributeSet`? That is meant to be just a staging area to accumulate attributes in between the `OnReportBegin()` and `OnReportEnd` calls.

You need to go straight to the real cache...

Reply:
agree, would update it, thanks

Reply:
done, thanks